### DISCUSSION TITLE
Long Term Memory Explanation

### DISCUSSION THREAD

**Discussion by atharvjairath** [facebookresearch/ParlAI#4159] [Q&A]
Hi
I Just wanted to know, how the BlenderBot 2 Uses the Long Term Memory. 

1. How a memory search is performed? What models and steps are used.
2. After finding a match, What kind of encoder is used to feed the summary to the final model in BB2?

Thank you!

**Comment by klshuster** [comment]
1. Memory is implemented as a DPR retrieval over a store of memories.
2. Memories are fed as knowledge and considered as such in a FiD-style model (see [FiD paper](https://arxiv.org/abs/2007.01282) for details)
